packages/mui-material/src/styles/createThemeNoVars.js

```diff
@@ -24,10 +24,13 @@ function createThemeNoVars(options = {}, ...args) {
     ...other
   } = options;
 
-  if (options.vars) {
+  if (
+    options.vars &&
+    options.generateThemeVars === undefined // check for `generateThemeVars` (created internally) to handle nested theme
+  ) {
     throw /* minify-error */ new Error(
       'MUI: `vars` is a private field used for CSS variables support.\n' +
-        'Please use another name.',
+        'Please use another name or follow the [docs](https://mui.com/material-ui/customization/css-theme-variables/usage/) to enable the feature.',
```

Comment:
Should we add a `#host-reference` comment for this line? See https://github.com/mui/material-ui/pull/45198